src/main/kotlin/com/viartemev/thewhiterabbit/publisher/TxPublisher.kt

```diff
@@ -0,0 +1,39 @@
+package com.viartemev.thewhiterabbit.publisher
+
+import com.rabbitmq.client.Channel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+class TxPublisher internal constructor(private val channel: Channel) {
+
+    suspend fun publishInTx(message: OutboundMessage) {
+
+        logger.debug { "publish in tx msg: $message" }
+
+        // todo Docs says -- "Invocations of Channel#basicPublish will eventually block if a resource-driven alarm  is in effect."
+        // todo shall we consider this case?
```

Comment:
It's a good point.
Based on docs (https://www.rabbitmq.com/alarms.html) it can happen in several cases, and we can do nothing with it. 
My suggestion is to add a timeout for basicPublish method and interrupt execution + set this timeout as an optional value for the publisher.